### DISCUSSION TITLE
Avalonia not applying external styles on TextBox children Properly

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#13415] [Q&A] [answered]
I have been trying to align the TextBox's Watermark & Content using the following styles, the selectors are working as the devtool shows, but the styling provided does not get applied to the specified controls : 

```axaml
<Style Selector="TextBox#VIEW_SearchBar /template/ TextPresenter">
  <Setter Property="VerticalAlignment" Value="Center" />
</Style>
<Style Selector="TextBox#VIEW_SearchBar /template/ TextBlock#PART_Watermark">
  <Setter Property="VerticalAlignment" Value="Center" />
</Style>
```

![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/755e2d1f-f0e4-4d68-8d46-0de897f763a9)

![image](https://github.com/AvaloniaUI/Avalonia/assets/39861216/e5d4deba-a8bc-4c9e-a0cf-31aca8ab6d60)

Manually setting the alignment in the dev tool works of course, but, the user shouldn't have to do that.
Is this a known issue or should this be converted to an issue?

**Comment by mgnslndh** [comment]
Why not set the VerticalContentAlignment of the TextBox? That will affect both the text and watermark.

**Reply by Mrcubix** [reply]
Cause that will probably affect the TextBox 'ss physical position, people asked for separate control over alignements long ago and other controls inheriting from TextBox got it

**Reply by mgnslndh** [reply]
TextBox.VerticalAlignment will affect the text box's physical position but VerticalContentAlignment affects the content, such as the watermark and text.

Look at the control theme for the Fluent TextBox (which I assume you are using according to your selectors). The properties you are setting are both using TemplateBinding on the VerticalContentAlignment property of the TextBox:

https://github.com/AvaloniaUI/Avalonia/blob/8da81002881485cea9b74086f128698033333505/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml#L144-L167

**Reply by Mrcubix** [reply, marked as answer]
Didn't see it in the list at first, but i'm also used to it existing in various other control and not working there.
Apparently it did something, it is more centered, but not centered (5 pixels up, 3 down).

![Image](https://i.imgur.com/Iqzvoix.png)

**Comment by timunie** [comment]
Have you checked the original style? TextBox has some special Padding applied which may be causing the weird offsets you see.
